frontend/src/components/Layout/AppBar/Buttons/TaskManagerButton.vue

```diff
@@ -70,104 +68,57 @@ const props = withDefaults(
-const scheduledTimeout = ref(0);
 const taskManager = taskManagerStore();
-
-const buttonColor = computed(() => {
-  return taskList.value.every((task) => {
-    return task.progress === 100;
-  })
-    ? 'success'
-    : undefined;
-});
-
-const showButton = computed(() => taskList.value.length > 0);
-
-/**
- *  Clear the tasks list
- */
-function clearState(): void {
-  taskList.value = [];
-}
-
-/**
- * Set a timeout to hide the button
- */
-function setTimeout(): void {
-  scheduledTimeout.value = window.setTimeout(clearState, props.timeout);
-}
-
-/**
- * Locally retrieve and parse the tasks list
- */
-function getTaskList(): void {
-  const list: Array<TaskInfo> = [];
-
-  for (const task of taskManager.tasks) {
-    switch (task.type) {
+let frozenTaskList: readonly TaskInfo[] = [];
+const runningTaskList = computed<TaskInfo[]>(() => {
+  return taskManager.tasks.map((t) => {
+    switch (t.type) {
       case TaskType.ConfigSync: {
-        list.push({
-          progress: undefined,
+        return {
+          progress: t.progress,
           textKey: 'appbar.tasks.configSync',
-          id: task.id
-        });
-
-        break;
+          id: t.id
+        };
       }
       case TaskType.LibraryRefresh: {
-        list.push({
-          progress: task.progress,
+        return {
+          progress: t.progress,
           textKey: 'appbar.tasks.scanningLibrary',
           textParams: {
-            library: task.data || ''
+            library: t.data || ''
           },
-          id: task.id
-        });
-
-        break;
+          id: t.id
+        };
       }
     }
-  }
+  });
+});
 
-  const taskIds = new Set(
-    list.map((task) => {
-      return task.id;
-    })
-  );
-  const finishedTasks: Array<TaskInfo> = [];
+const allCompleted = computed(() =>
+  runningTaskList.value.every((t) => t.progress === 100)
+);
 
-  for (const task of taskList.value) {
-    if (!taskIds.has(task.id)) {
-      task.progress = 100;
-      finishedTasks.push(task);
-    }
-  }
+const buttonColor = computed(() => {
+  return allCompleted.value ? 'success' : undefined;
+});
 
-  taskList.value = [...list, ...finishedTasks];
-}
+const taskList = computed(() => {
+  return menu.value && allCompleted.value
+    ? frozenTaskList
+    : runningTaskList.value;
+});
+
+const showButton = computed(() => taskList.value.length > 0);
 
 watch(
-  taskManager.tasks,
-  () => {
-    getTaskList();
-  },
+  () => props.timeout,
+  () => (taskManager.timeout = props.timeout),
   { immediate: true }
 );
 
-watch(buttonColor, () => {
-  window.clearTimeout(scheduledTimeout.value);
-
-  if (buttonColor.value && !menu.value) {
-    setTimeout();
-  }
-});
-
-watch(menu, () => {
-  if (menu.value) {
-    window.clearTimeout(scheduledTimeout.value);
-  } else if (buttonColor.value) {
-    setTimeout();
+watch([menu, allCompleted], () => {
+  if (menu.value && allCompleted.value) {
+    frozenTaskList = Object.freeze([...toRaw(runningTaskList.value)]);
```

Comment:
@aweebs I guess this should save a few CPU cycles?

No need to do this calculation when the menu goes to the closed state or all tasks are not completed, just for this exact situation. The result its the same, but I feel its unnecessary, unless I'm missing something

Reply:
Not sure there's much CPU cycle savings happening here, since this function still has to be called when `menu` changes right now but it just won't do anything. And if we're truly worried about CPU cycles I think we'd be in a dire spot using JavaScript :D

My argument here would just be readability/understandability. The dependency of this getting set is really only on `allCompleted`.